src/main/java/org/candlepin/subscriptions/resource/CapacityResource.java

```diff
@@ -156,10 +169,15 @@ public CapacityReport getCapacityReport(ProductId productId, @NotNull Granularit
 
         List<CapacitySnapshot> result = new ArrayList<>();
         OffsetDateTime next = OffsetDateTime.from(start);
+
+        /* Walk through the timeline by the smallest unit of granularity.  Create capacity snapshots at each
+         * point on the timeline. */
         while (next.isBefore(end) || next.isEqual(end)) {
-            result.add(createCapacitySnapshot(clock.startOfDay(next), matches));
-            next = clock.startOfDay(next.plus(offset));
+            OffsetDateTime timelinePoint = clock.tick(next, productGranularity);
+            result.add(createCapacitySnapshot(timelinePoint, matches));
+            next = clock.startOfHour(next.plus(offset));
```

Comment:
This is really the critical section that I want people to think about.  I'm a little torn around wondering if `next` should be incremented based on a `switch` statement rather than just by every hour.

E.g.

```
switch (productGranularity):
    case HOURLY:  next = clock.startOfHour(next.plus(offset));
    case DAILY: next = clock.startOfDay(next.plus(offset));
...
```

I'm kind of leaning that way at the moment, but I go back and forth.